game-core/src/main/java/games/strategy/engine/data/UnitFactory.java

```diff
@@ -1,10 +1,8 @@
 package games.strategy.engine.data;
 
-import java.io.Serializable;
-
 /**
  * Factory for creating instances of {@link Unit}.
  */
-public interface IUnitFactory extends Serializable {
+public interface UnitFactory {
```

Comment:
You should annotate this interface with the `@FunctionalInterface` annotation to explicitly indicate that this class can be implemented by a lambda.

Reply:
I actually did that originally and then reverted the change.  My reasoning was that I didn't want to imply to a future implementer that `UnitFactory` _has_ to be functional (e.g. one could easily envision a caching `UnitFactory` that has mutable state).  So I wasn't sure the intent of this interface was in the spirit of `@FunctionalInterface`.

But, I guess `UnitFactory` is really no different than a `Supplier`, which I probably would have used if `createUnit()` didn't require parameters.

38af157a3